pymatgen/command_line/bader_caller.py

```diff
@@ -179,49 +191,63 @@ def __init__(
-                atom_chgcars = [Chgcar.from_file(f"BvAt{idx + 1:04}.dat") for idx in range(len(self.chgcar.structure))]
-
-                atomic_densities = []
-                # For each atom in the structure
-                for _site, loc, chg in zip(self.chgcar.structure, self.chgcar.structure.frac_coords, atom_chgcars):
-                    # Find the index of the atom in the charge density atom
-                    index = np.round(np.multiply(loc, chg.dim))
-
-                    # Find the shift vector in the array
-                    shift = (np.divide(chg.dim, 2) - index).astype(int)
-
-                    # Shift the data so that the atomic charge density to the center for easier manipulation
-                    shifted_data = np.roll(chg.data["total"], shift, axis=(0, 1, 2))
-
-                    # Slices a central window from the data array
-                    def slice_from_center(data, x_width, y_width, z_width):
-                        x, y, z = data.shape
-                        start_x = x // 2 - (x_width // 2)
-                        start_y = y // 2 - (y_width // 2)
-                        start_z = z // 2 - (z_width // 2)
-                        return data[
-                            start_x : start_x + x_width,
-                            start_y : start_y + y_width,
-                            start_z : start_z + z_width,
-                        ]
-
-                    def find_encompassing_vol(data: np.ndarray):
-                        # Find the central encompassing volume which holds all the data within a precision
-                        total = np.sum(data)
-                        for idx in range(np.max(data.shape)):
-                            sliced_data = slice_from_center(data, idx, idx, idx)
-                            if total - np.sum(sliced_data) < 0.1:
-                                return sliced_data
-                        return None
-
-                    dct = {
-                        "data": find_encompassing_vol(shifted_data),
-                        "shift": shift,
-                        "dim": self.chgcar.dim,
-                    }
-                    atomic_densities.append(dct)
-                self.atomic_densities = atomic_densities
+                self._parse_atomic_densities()
+
+    @deprecated("parse_atomic_densities deprecated and would be removed on 26Feb2024.")
+    def _parse_atomic_densities(
+        self,
+    ):
+        def slice_from_center(data, x_width, y_width, z_width):
+            """Slices a central window from the data array."""
+            x, y, z = data.shape
+            start_x = x // 2 - (x_width // 2)
+            start_y = y // 2 - (y_width // 2)
+            start_z = z // 2 - (z_width // 2)
+            return data[
+                start_x : start_x + x_width,
+                start_y : start_y + y_width,
+                start_z : start_z + z_width,
```

Comment:
For some reason unknown to me, `ruff-format` seems against removal of `whitespace before ':'`? I'm a little confused.

Reply:
What about this? @janosh It seems `flake8` and `ruff-format` go against each other...

Reply:
we don't actually use `flake8` so `ruff-format` determines formatting

Reply:
Okay thanks... Sorry didn't notice that.